hw/ip/entropy_src/doc/_index.md

```diff
@@ -46,28 +50,41 @@ However, as long as the function provided by the register set supports all of th
-The initialization step can be done at anytime while the block is enabled and running.
-Before setting the `ENABLE` bit, the LFSR seed input should be initialized, using the {{< regref "ES_SEED" >}} register.
+The first step is initialization and enabling.
+Before setting the `ENABLE` field, the LFSR seed input should be initialized, using the {{< regref "ES_SEED" >}} register.
 The {{< regref "ES_SEED" >}} register will transfer its value according to the initialization sequence.
 After the block is enabled and initialized, entropy bits will be collected up indefinitely until disabled.
+If analog mode is desired, the `ANA_SRC_EN` bit should be set first.
+Once the `es_ana_src_ok` interrrupt or status bit is active, then the `ENABLE` field can be set to enable analog mode.
 
 It is assumed that any entropy source will drop bits.
-This means that the FIFO used to collect up the entropy bits will fill, and until the firmware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
-The process of filling and draining the FIFO is the same, independent of what speed the entropy generation rate is at.
+This means that the FIFO used to collect up the entropy bits will fill, and until the firmware or hardware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
+The process of filling and draining the FIFOs is the same, independent of what speed the entropy generation rate is at.
 
-Once the `ENTROPY_RDY` status bit is set, the firmware will read the {{< regref "ES_ENTROPY" >}} register as many times as needed to get the required entropy bits, typically 128 bits or 256 bits at a time.
+Once the `es_entropy_valid` status bit is set, the firmware will read the {{< regref "ES_ENTROPY" >}} register as many times as needed to get the required entropy bits, typically 128 bits or 256 bits at a time.
 For ease of firmware operation, the status bit does not need to be reset.
 A read of the {{< regref "ES_THRESH" >}} register could be done to verify how many entries are in the FIFO at the current time.
 The {{< regref "ES_THRESH" >}} register will determine when the status bit is set, or when the interrupt will assert if enabled.
-For this revision, this same {{< regref "ES_THRESH" >}} register has a range of between 1 and 32 entries in the FIFO.
-An RTL parameter can be set to make this range smaller and save on gate count.
+For this revision, this same {{< regref "ES_THRESH" >}} register has a range of between 1 and 4 entries in the TLUL FIFO.
+
+For hardware consumption of entropy, if the HWIF FIFO is not empty, and the entropy consuming hardware is ready, then entropy bits will be transferred over the hardware entropy interface.
+
+Health tests, when enabled in the {{< regref "ES_CONF" >}} register, will run continuously.
+Two tests are implemented in hardware, the repetition test, and the adaptive proportion test.
```

Comment:
you may want to define the acronyms here for later reference of these tests (I think the acronyms are used throughout the regfile).

Reply:
good point, added